test/junit/scala/reflect/FieldAccessTest.scala

```diff
@@ -0,0 +1,25 @@
+package scala.reflect
+
+import org.junit.Assert._
+import org.junit.Ignore
+import org.junit.Test
+
+class FieldAccessTest {
+
+  class TestClass {
+    private val x = 123
+    // Uncommenting the following line would make the test fail
+    () => x
+  }
+
+  /** scala/bug#9306 */
+  @Test
+  def testFieldAccess(): Unit = {
+    import scala.reflect.runtime.{universe => ru}
+    val mirror = ru.runtimeMirror(getClass.getClassLoader)
```

Comment:
i.e., `scala.reflect.runtime.currentMirror`

Reply:
I'll give it a a try...